frontends/web/src/routes/buy/pocket.tsx

```diff
@@ -101,43 +103,47 @@ export const Pocket = ({ code }: TProps) => {
   };
 
   const handleRequestAddress = (message: RequestAddressV0Message) => {
-    signing = true;
-    const addressType = message.withScriptType ? String(message.withScriptType) : '';
-    const withMessageSignature = message.withMessageSignature ? message.withMessageSignature : '';
-    signAddress(
-      addressType,
-      withMessageSignature,
-      code)
-      .then(response => {
-        signing = false;
-        if (response.success) {
-          sendAddress(response.address, response.signature);
-        } else {
-          if (response.errorCode !== 'userAbort') {
-            alertUser(t('unknownError', { errorMessage: response.errorMessage }));
-            console.log('error: ' + response.errorMessage);
+    requestKeystore(code, async () => {
+      signing = true;
+      const addressType = message.withScriptType ? String(message.withScriptType) : '';
+      const withMessageSignature = message.withMessageSignature ? message.withMessageSignature : '';
+      signAddress(
```

Comment:
the callback is marked `async`, so you should use `await` here instead of `...then()`. this or remove the `async` keysword as it is only needed if you want to make use of `await`